### ISSUE TITLE
Controller broken

### ISSUE THREAD

**Issue by sonicfelipemt** [SRB2-Mobile/SRB2-Android#73]
I'm on ZenFone 8 + DualSense controller. 

I can now use the joystick to move, and press R1 to aim-lock. The problem appears when trying to press the face buttons (square, X, triangle and circle and d-pad). They work sometimes but won't fully register when I hold them down (jump or spin only work sometimes and as short press, when I press them repeatedly). 
At first it appears player 2 is auto-connected to it, so I disable it for P2 and enable for P1.
Then it works in the broken way I described b4. 
Other games register only as 1 controller and have no face button problems like that.

**Comment by jaykozatt** [comment]
This post, along with information shared on other issues, allowed me to get a breakthrough on this matter. So I'll breakdown what I found out for anyone that comes along with this issue:

* **The game expects X-Input controllers.** If your controller uses D-Input, it won't work.
  * Mine had several modes: two of them were advertised for Android in the controller's manual, and another one of them for PC. The Android ones were a V3 emulation that mapped controller input to touch input, and the other was D-Input. The PC one despite being advertised as for use in PC, was X-Input, which is precisely what this port was looking for.
* If your controller outputs using X-Input mapping, then the game will behave as *sonicfelipemt* describes. 
  * The controller seems to "work" but it only gives you control over the secondary character if your playing for example with the Sonic & Tails tag team. This is because the default configuration sets up "player 2" to be controlled using a gamepad. In my game, my controller wasn't listed when selecting a gamepad, despite it definitely having control over player 2. So when I selected "None", I could no longer control player 2 with it afterwards, nor could I restore its previous setting from within the app.
  * The above led me to the config text file found in the SRB2 folder, in which I found the lines `gamepad=0` and `gamepad2=0`. And changing the first one to `gamepad=1` enabled my controller once again, this time giving me control over player 1, and the whole UI without issue. Therefore, this is clearly just an UI problem. The correct functionality is there.

And that's basically how I fixed the issue for me.

**TL;DR:** Verify that your controller maps its inputs using X-Input and not something else. And go to the config text file found in the SRB2 folder, and change the line `gamepad=0` and `gamepad2=1` to `gamepad=1` and `gamepad2=0` so that the controller input maps correctly to player 1.
